### PR TITLE
MGMT-8444: Update edit hostname dialog

### PR DESCRIPTION
**Pull request by rawagner**
change_hostname
Screenshot from 2022-01-04 14-04-33

Screenshot from 2022-01-04 14-04-21
Screenshot from 2022-01-04 14-06-23

@jkilzi FYI this also updates the dialog in OCM

ACM PR https://github.com/open-cluster-management/console/pull/1065

### CODE REVIEW

**Comment by jkilzi** [comment]
@rawagner lgtm, but...
@nirfarkas are you aware of this change? Are you happy with it?

**Comment by nirfarkas** [comment]
Not really ):
Why not change to using this?
https://github.com/patternfly/patternfly/issues/4017
It's interactive takes less space and is more elegant.

**Comment by ShiranHi** [comment]
Using this rich helper text: https://marvelapp.com/prototype/haechg7/screen/84022338
Also in the Jira ticket: https://issues.redhat.com/browse/MGMT-8444

File: src/common/components/hosts/EditHostForm.tsx [hunk 1]
```diff
@@ -104,14 +108,26 @@ const EditHostForm: React.FC<EditHostFormProps> = ({
...
                 ref={hostnameInputRef}
-                helperText="This name will replace the original discovered hostname after installation."
+                helperText={
+                  <List className="edit-hostname__help-text">
+                    <ListItem>Must be at least 5 characters</ListItem>
+                    <ListItem>Use lower-case letters, digits from 0 to 9 or hyphen (-)</ListItem>
+                    <ListItem>Cannot start or end with a hyphen (-)</ListItem>
+                    <ListItem>Cannot be the word "localhost"</ListItem>
+                  </List>
+                }
```
Comment (jkilzi) [inline src/common/components/hosts/EditHostForm.tsx:130]: @ShiranHi Patternfly already provides dynamic helper texts.
Please see this example: https://www.patternfly.org/v4/components/helper-text#dynamic-list

Comment (jkilzi) [inline src/common/components/hosts/EditHostForm.tsx:130]: @rawagner, @ShiranHi Patternfly already provides dynamic helper texts.
Please see this example: https://www.patternfly.org/v4/components/helper-text#dynamic-list

**Review by jkilzi — CHANGES_REQUESTED** [review]

File: src/common/components/hosts/EditHostForm.tsx (hunk 1 above)
Comment (jkilzi) [reply src/common/components/hosts/EditHostForm.tsx:130]: We have already a task in order to align the PF spec in this regard.
See: https://issues.redhat.com/browse/MGMT-8275

**Comment by rawagner** [comment]
allright, i misunderstood what this rich text component should represent. Sorry about that.
PR is now updated (pics included). 
@ShiranHi can you take a look again ? I've also added/changed some validation messages to properly reflect what we validate

**Comment by nirfarkas** [comment]
@rawagner are the pics here updated?

**Comment by ShiranHi** [comment]
Thanks @rawagner it looks good, I am happy to see it :)
Just wrapping up our call: "Must be unique" will be the first bullet point and we use the custom check and cross marks icons, as in [PF](https://www.patternfly.org/v4/components/helper-text/#dynamic-list).

**Comment by jkilzi** [comment]
@rawagner we can really make use of a GIF here showing how it behaves

**Review by jkilzi — DISMISSED** [review]

**Comment by rawagner** [comment]
@nirfarkas I did not press `Update comment` ..so ehm no, screenshots were not updated :) 

Anyways I added a gif as @jkilzi suggested.

@ShiranHi 
- Moved `Must be unique` to top
- Added 'default' state when there is no value in the field. Note that the default state does not have bullets but minus icon since there is no option to have bullets in PF react

**Review by jkilzi — APPROVED** [review]

**Pull request merged**